### ISSUE TITLE
Ease the no-confusing-arrow rule

### ISSUE THREAD

**Issue by madsgodvinjensen** [trustpilot/node-eslint-config#33]
I propose we ease the no-confusing-arrow rule a bit and use the follow setting:

`'no-confusing-arrow': [1, { allowParens: true }]`

That would mean that this is a warning:

```javascript
const a = () => b() ? 'yes' : 'no';
```

... but this is accepted:

```javascript
const a = () => (b() ? 'yes' : 'no');
```

What do you say?

**Comment by thomasthiebaud** [comment]
Looks reasonable 👍

**Comment by DanMeakin** [comment]
I personally dislike the rule; I don't get how the arrow operator is confusing. So my preference would be to disable this rule. However, if others think it is a useful rule, then I think permitting parens would be a useful loosening of it.

**Comment by TheLaz** [comment]
LGTM 👍

**Comment by madsgodvinjensen** [comment]
I don't find arrow functions confusing either - but I figured the rule had been put in there for a reason. In the eslint docs, the difference between these following two lines is essentially what the rule is trying to make clearer:

```javascript
var x = a => 1 ? 2 : 3;
var x = a <= 1 ? 2 : 3;
```

To me that example is a bit too contrived and I don't think it's an actual issue. I would also prefer to remove it completely.

**Comment by madsgodvinjensen** [comment]
Fixed in release [v3.2.1](https://github.com/trustpilot/node-eslint-config/releases/tag/v3.2.1)
